plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -775,6 +775,14 @@ func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercach
 	if !fit {
 		return fit, err
 	}
+	// TODO(HaiyangDING): Make a proper fix for it.
+	allocatable := node.Status.Allocatable
+	allowedPodNumber := allocatable.Pods().Value()
+	if int64(len(nodeInfo.Pods()))+1 > allowedPodNumber {
+		return false,
+			newInsufficientResourceError(podCountResourceName, 1, int64(len(nodeInfo.Pods())), allowedPodNumber)
+	}
+
```

Comment:
aah - ok. let's leave as it - I will fix that in my PR where I'm changing it :)
With https://github.com/kubernetes/kubernetes/pull/24598 - it should be easy to do :)